syncr_backend/util/drop_util.py

```diff
@@ -151,17 +154,71 @@ def get_drop_metadata(
 
     if metadata is None:
         logger.debug("drop metadata not on disk, getting from network")
+        args = {
+            'drop_id': drop_id,
+            'drop_version': version,
+        }
         metadata = send_requests.do_request(
             request_fun=send_requests.send_drop_metadata_request,
             peers=peers,
-            fun_args={'drop_id': drop_id},
+            fun_args=args,
         )
 
         metadata.write_file(is_latest=True, metadata_location=metadata_dir)
 
     return metadata
 
 
+def verify_version(drop_metadata: DropMetadata) -> None:
```

Comment:
Can you pass peers in here as an optional? Then pass the peers in the recursive call. So we don't re look it up every call.